libraries/chain/transaction_context.cpp

```diff
@@ -176,11 +178,22 @@ namespace eosio { namespace chain {
          billing_timer_exception_code = leeway_deadline_exception::code_value;
       }
 
-      billing_timer_duration_limit = _deadline - start;
+      // Possibly limit deadline to subjective max_transaction_time
+      if( max_transaction_time_subjective != fc::microseconds::maximum() && (start + max_transaction_time_subjective) <= _deadline ) {
+         _deadline = start + (max_transaction_time_subjective == fc::microseconds::maximum() ?
+                              fc::hours(24*7*52) : max_transaction_time_subjective);
+         billing_timer_exception_code = tx_cpu_usage_exceeded::code_value;
+      }
+
+      // Possibly limit deadline to caller provided wall clock block deadline
+      if( block_deadline < _deadline ) {
+         _deadline = block_deadline;
+         billing_timer_exception_code = block_cpu_usage_exceeded::code_value;
```

Comment:
Shouldn't `block_deadline` correspond to `deadline_exception`?  I thought `block_cpu_usage_exceeded` means that the billed time exceeds the objective block cpu limit.

Reply:
I was thinking the user would not care the difference. But our exception does say `billable`, so I'll go back to having deadline for this. It probably is useful to have diff exceptions for these two cases.